tests/executors/test_kubernetes_executor.py

```diff
@@ -182,33 +182,59 @@ def test_delete_pod_404_not_raised(self, mock_watcher, mock_client, mock_kube_cl
         mock_delete_namespace.assert_called_with(pod_id, namespace, body=mock_client.V1DeleteOptions())
 
 
-class TestKubernetesExecutor(unittest.TestCase):
+class TestKubernetesExecutor:
     """
     Tests if an ApiException from the Kube Client will cause the task to
     be rescheduled.
     """
 
-    def setUp(self) -> None:
+    def setup_method(self) -> None:
         self.kubernetes_executor = KubernetesExecutor()
         self.kubernetes_executor.job_id = "5"
 
-    @unittest.skipIf(AirflowKubernetesScheduler is None, 'kubernetes python package is not installed')
+    @pytest.mark.skipif(
+        AirflowKubernetesScheduler is None, reason='kubernetes python package is not installed'
+    )
+    @pytest.mark.parametrize(
+        'reason, should_requeue',
+        [
+            ('Forbidden', True),
+            ('fake-unhandled-reason', True),
+            ('Unprocessable Entity', False),
+            ('BadRequest', False),
```

Comment:
I think I'd prefer to see the status code here instead, particularly since we aren't using the `reason` for anything. Then we can also drop the map from inside the test body too.

If we do need both, I think the mapping should happen here anyways.

Reply:
yeah that's the right call.